### ISSUE TITLE
Doesn't work with apple silicone

### ISSUE THREAD

**Issue by duxbuse** [terraform-google-modules/terraform-google-cloud-storage#243]
### TL;DR

Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/hashicorp/null v2.1.2 does not have a package available for your current platform,
│ darwin_arm64.
│ 
│ Provider registry.terraform.io/hashicorp/random v2.3.1 does not have a package available for your current platform,
│ darwin_arm64.
|
│ Provider releases are separate from Terraform CLI releases, so not all providers are available for all platforms.

### Expected behavior

I should be able to run `terraform init` with no problems of arm64 architecture.

### Observed behavior

Both the null and random resource types have become built in functions. As a result these providers are deprecated and do not and never will support `arm64`. So this module ideally updates to use the built in functions allowing arm64 devices to use this module.

### Terraform Configuration

```hcl
module "gcs_buckets" {
  source            = "terraform-google-modules/cloud-storage/google"
  version           = "~> 2.2"
}
```

### Terraform Version

```sh
Terraform v1.4.2
on darwin_arm64
```
